sycl/include/sycl/builtins.hpp

```diff
@@ -25,6 +25,17 @@ namespace detail {
 template <class T, size_t N> vec<T, 2> to_vec2(marray<T, N> x, size_t start) {
   return {x[start], x[start + 1]};
 }
+template <class T, size_t N> vec<T, N> to_vec(marray<T, N> x) {
+  vec<T, N> vec;
+  std::memcpy(&vec, &x, sizeof(x));
```

Comment:
I'm a bit concerned that using `&vec` and `&x` might cause trouble if their underlying implementations at some point are not simply the data. Would that be ever possible?

Reply:
Re-wrote to for-loop here to avoid possible alignment issues as incorrect alignment leads to UB.